From 93ff3c3ceff701a04d8f585cd4a211fa2567bc74 Mon Sep 17 00:00:00 2001
From: Bhushan Shah <bhush94@gmail.com>
Date: Thu, 4 Mar 2021 13:25:40 +0530
Subject: [PATCH] platforms/drm: inhibit the renderloop when screen is turned
 off

Otherwise we will continue using gpu resources even when we can not show
any content on screen.

(cherry picked from commit debea631f38a37580921311ab0cf316105873da4)
---
 plugins/platforms/drm/drm_output.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/plugins/platforms/drm/drm_output.cpp b/plugins/platforms/drm/drm_output.cpp
index fddf6765d..19e6eed04 100644
--- a/plugins/platforms/drm/drm_output.cpp
+++ b/plugins/platforms/drm/drm_output.cpp
@@ -582,6 +582,7 @@ void DrmOutput::dpmsFinishOn()
     if (!m_gpu->atomicModeSetting()) {
         m_crtc->blank();
     }
+    m_renderLoop->uninhibit();
     if (Compositor *compositor = Compositor::self()) {
         compositor->addRepaintFull();
     }
@@ -597,6 +598,7 @@ void DrmOutput::dpmsFinishOff()
     } else {
         waylandOutput()->setDpmsMode(toWaylandDpmsMode(DpmsMode::Off));
     }
+    m_renderLoop->inhibit();
 }
 
 bool DrmOutput::dpmsLegacyApply()
-- 
GitLab

